plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStore.java

```diff
@@ -71,13 +71,28 @@ public S3BlobStore(Settings settings, AmazonS3 client, String bucket, @Nullable
         }
 
         this.numberOfRetries = maxRetries;
-        if (!client.doesBucketExist(bucket)) {
-            if (region != null) {
-                client.createBucket(bucket, region);
-            } else {
-                client.createBucket(bucket);
+
+        int retry = 0;
+        while (retry <= this.numberOfRetries) {
+            try {
+                if (!client.doesBucketExist(bucket)) {
+                    if (region != null) {
+                        client.createBucket(bucket, region);
+                    } else {
+                        client.createBucket(bucket);
+                    }
+                }
+                break;
+            } catch (AmazonClientException e) {
+                if (shouldRetry(e) && retry < this.numberOfRetries) {
```

Comment:
> retry < this.numberOfRetries is implied here due to the outer check.
I don't think internal retry < numberOfRetries is necessary, but we have done it for our retries in this file (https://github.com/elastic/elasticsearch/blob/master/plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/blobstore/DefaultS3OutputStream.java#L163). I prefer to make them stay in the same way for now, a refactor for this retry logic could be done. Let me know what you think.  

> I also wonder if we should log TRACE/DEBUG issues for this.
Agree, will add a "log trace" for this.